lib/properties.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+const EventEmitter = require('events').EventEmitter;
+
+class Layer extends EventEmitter {
+  constructor(parent, namespace) {
+    super();
+
+    this.parent = parent;
+    this.namespace = namespace;
+  }
+}
+
+/**
+ * A static Layer wraps an Object of properties with an optional namespace
+ */
+class Static extends Layer {
+  constructor(parent, properties, namespace) {
+    super(parent, namespace);
+    this.properties = properties;
+  }
+}
+
+/**
+ * A Dynamic layer wraps a Source with an optional namespace and forwards its
+ * update events to a parent Properties object.
+ */
+class Dynamic extends Layer {
+  constructor(parent, source, namespace) {
+    super(parent, namespace);
+
+    this.source = source;
+    source.on('update', () => parent.build());
+  }
+
+  get properties() {
+    return this.source.properties;
+  }
+}
+
+/**
+ * A View aggregates the initialized events of multiple sources and safely
+ * forwards their update events to a parent Properties object.
+ */
+class View extends Layer {
```

Comment:
Does `View` need to extend `Layer`? It doesn't look like it's emitting events or leveraging the namespace concept that `Layer` provides.